File: hanapbahay-backend/Models/Entities/Property.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace hanapbahay_backend.Models.Entities;

[Index(nameof(City))]
[Index(nameof(Province), nameof(City), nameof(ZipCode))]
[Index(nameof(Status), nameof(City), nameof(PropertyType))]
public class Property
{
    public int Id { get; set; }

    [Required]
    public Guid LandlordId { get; set; }
    public User Landlord { get; set; } = null!;

    [Required, StringLength(150, MinimumLength = 3)]
    public string Title { get; set; } = null!;

    [StringLength(2000)]
    public string? Description { get; set; }

    public PropertyType PropertyType { get; set; }

    [Required]
    public string Province { get; set; } = null!;
    [Required]
    public string City { get; set; } = null!;
    public string? Barangay { get; set; }
    public string? ZipCode { get; set; }

    public string? TargetLocation { get; set; }
    public string? Landmark { get; set; }

    [Precision(18, 2)]
    public decimal MonthlyPrice { get; set; }

    public byte? MaxPersons { get; set; }

    public DateTime? MoveInDate { get; set; }

    public ListingStatus Status { get; set; } = ListingStatus.Draft;

    public bool IsDeleted { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation
    public ICollection<PropertyAmenity> PropertyAmenities { get; set; } = new List<PropertyAmenity>();
    public ICollection<Media> Media { get; set; } = new List<Media>();
}

File: hanapbahay-frontend/src/types/property.ts
export const propertyTypeOptions = [
  { label: "House", value: 0 },
  { label: "Condo", value: 1 },
  { label: "Apartment", value: 2 },
  { label: "Bed Spacer", value: 3 },
  { label: "Dorm", value: 4 },
  { label: "Room", value: 5 },
] as const;

export type PropertyTypeValue = (typeof propertyTypeOptions)[number]["value"];

export const listingStatusOptions = [
  { label: "Draft", value: 0 },
  { label: "Active", value: 1 },
  { label: "Paused", value: 2 },
  { label: "Reserved", value: 3 },
  { label: "Rented", value: 4 },
  { label: "Removed", value: 5 },
] as const;

export type ListingStatusValue = (typeof listingStatusOptions)[number]["value"];

export interface PropertyMedia {
  id: number;
  url: string;
  order: number;
  isCover: boolean;
}

export interface Property {
  id: number;
  landlordId: string;
  landlordDisplayName: string;
  title: string;
  description?: string | null;
  propertyType: PropertyTypeValue;
  province: string;
  city: string;
  barangay?: string | null;
  zipCode?: string | null;
  targetLocation?: string | null;
  landmark?: string | null;
  monthlyPrice: number;
  maxPersons?: number | null;
  moveInDate?: string | null;
  status: ListingStatusValue;
  createdAt: string;
  amenityCodes: string[];
  media: PropertyMedia[];
}

export interface CreatePropertyPayload {
  title: string;
  description?: string;
  propertyType: PropertyTypeValue;
  province: string;
  city: string;
  barangay?: string;
  zipCode?: string;
  targetLocation?: string;
  landmark?: string;
  monthlyPrice: number;
  maxPersons?: number;
  moveInDate?: string;
  status: ListingStatusValue;
  amenityCodes: string[];
  coverImage: File;
  galleryImages: File[];
}

export interface UpdatePropertyPayload
  extends Omit<CreatePropertyPayload, "coverImage" | "galleryImages"> {
  newCoverImage?: File;
  newGalleryImages: File[];
  removeImageIds: number[];
}

export interface FormState {
  title: string;
  description: string;
  propertyType: string;
  province: string;
  city: string;
  barangay: string;
  zipCode: string;
  targetLocation: string;
  landmark: string;
  monthlyPrice: string;
  maxPersons: string;
  moveInDate: string;
  status: string;
  amenityCodes: string;
}
